Megadeth/AppSettings.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace Megadeth;
+
+public class AppSettings
+{
+    public int ApiId { get; init; }
+    public string ApiHash { get; init; }
+    public  string PhoneNumber { get; init; }
+    public  string ApplicationVersion { get; set; }
+
+    public  bool AuthNeeded { get; set; }
+    public  bool PasswordNeeded { get; set; }
+    public readonly ManualResetEventSlim ReadyToAuthenticate = new();
```

Comment:
I don't think that these three, `AuthNeeded`, `PasswordNeeded` and `ReadyToAuthenticate`, belong to the application settings. Let's move them back to the service layer?